scripts/azureml-assets/azureml/assets/publish_utils.py

```diff
@@ -190,6 +190,12 @@ def validate_and_prepare_pipeline_component(
             logger.print(f"Workspace asset URI was used, using component from registry {registry}")
             registry = registry_name
 
+        if not version:
+            logger.log_error(
+                f"Component {name} has invalid version {version}. Please ensure a valid version is specified."
```

Comment:
Consider saying something different if `label` is set by `get_parsed_details_from_asset_uri`.